src/hooks/useDrafts/useDrafts.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react'
+import { addOrUpdateDraft, clearDrafts, draft, drafts, removeDraft } from './utils'
+
+type CommonDraftProps = { id: string; recentlyUpdatedAt: string }
+
+export const useDraft = <T extends object>(state: T) => {
+  const [draftsState, setDraftsState] = useState<(T & CommonDraftProps)[]>([])
+
+  const getInitialDrafts = async () => {
+    const currentDrafts = await drafts()
+    setDraftsState(currentDrafts)
+  }
+
+  const getSingleDraft = async (draftId: string) => {
+    const singleDraft = await draft(draftId)
+    return singleDraft
+  }
+
+  useEffect(() => {
+    getInitialDrafts()
+  }, [])
+
+  const createOrSaveDraft = async (draftId: string) => {
+    await addOrUpdateDraft(draftId, { ...state })
+    await getInitialDrafts()
+  }
+
+  const discardDraft = async (draftId: string) => {
+    await removeDraft(draftId)
+    await getInitialDrafts()
+  }
+
+  const discardAllDrafts = async () => {
+    clearDrafts()
+    await getInitialDrafts()
+  }
+
+  return {
+    drafts: draftsState,
+    createOrSaveDraft,
+    getSingleDraft,
+    discardDraft,
+    discardAllDrafts,
```

Comment:
```suggestion
   getDraft: getSingleDraft,
    removeDraft: discardDraft,
    removeAllDrafts: discardAllDrafts,
```

This could be something like this so the names from the utils don't clash and so we get a nicer API on the hook that's what we are going to be consuming most of the time